File: kinotic-js/workspace/packages/system-api/src/api/model/workload/VmNodeState.ts
import type { VmNodeStatusType } from '@/api/model/workload/VmNodeStatusType'

/**
 * The reconciled state of a node: the shape both the intent and the observation take, so a node is
 * in its desired state exactly when the two are equal. Intent is that the node takes workloads;
 * observation is what the node reports it can do.
 */
export interface VmNodeState {
    /** Whether the node takes workloads. */
    phase: VmNodeStatusType
}

File: kinotic-system-api/src/main/java/org/kinotic/system/api/model/workload/VmNodeState.java
package org.kinotic.system.api.model.workload;

/**
 * The reconciled state of a {@link VmNode}: the shape both the intent and the observation take, so a
 * node is in its desired state exactly when the two are equal. Intent is that the node takes
 * workloads; observation is what the node reports it can do.
 *
 * @param phase whether the node takes workloads
 */
public record VmNodeState(VmNodeStatusType phase) {
}
